client/src/pages/metadata-query/__tests__/date-query.test.tsx

```diff
@@ -0,0 +1,71 @@
+import { describe, expect, test, vi } from 'vitest';
+import { render, screen, fireEvent} from '@testing-library/react';
+
+import React from 'react';
+
+import DateQuery from '@/pages/metadata-query/date-query';
+
+describe('Test DateQuery', () => {
+  const mockName = 'date name';
+  const mockDescription = 'date description';
+
+  test('Basic Rendering', () => {
+    const validatorMock = vi.fn(() => 'this is valid');
+    const handleQueryValidMock = vi.fn();
+    const handleQueryInvalidMock = vi.fn();
+    render(
+      <DateQuery
+        queryName={mockName}
+        description={mockDescription}
+        validator={validatorMock}
+        handleQueryValid={handleQueryValidMock}
+        handleQueryInvalid={handleQueryInvalidMock}
+      />
+    );
+    expect(screen.getByText('date name')).exist;
+  });
+
+  test('Changing to and from to valid dates, calls handleQueryValid function', () => {
+    const validatorMock = vi.fn(() => 'this is valid');
+    const handleQueryValidMock = vi.fn();
+    const handleQueryInvalidMock = vi.fn();
+    render(
+      <DateQuery
+        queryName={mockName}
+        description={mockDescription}
+        validator={validatorMock}
+        handleQueryValid={handleQueryValidMock}
+        handleQueryInvalid={handleQueryInvalidMock}
+      />
+    );
+    const dateFrom = screen.getByTestId('date-from');
+    const dateTo = screen.getByTestId('date-to');
+
+    fireEvent.change(dateFrom, { target: { value: '2023-01-01' } });
```

Comment:
I would switch this to use userEvent:
https://testing-library.com/docs/ecosystem-user-event/

And again, test the behavior